tests/tel2puml/test_tel2puml_types.py

```diff
@@ -0,0 +1,86 @@
+"""Tests for the pydantic models within tel2puml_types.py"""
+
+import tempfile
+
+import pytest
+from pydantic import ValidationError
+
+from tel2puml.tel2puml_types import OtelToPVArgs, PvToPumlArgs
+
+
+@pytest.mark.parametrize("command", ["otel2pv", "otel2puml"])
+def test_otel_to_pv_args(command: str) -> None:
+    """Tests for the pydantic model OtelToPVArgs"""
+
+    # Test 1: Valid inputs for otel2pv
+    with tempfile.NamedTemporaryFile(suffix=".yaml") as tmp_file:
+        args = OtelToPVArgs(config_file=tmp_file.name, command=command)
+        assert str(args.config_file) == tmp_file.name
+        assert args.ingest_data is True
+        assert args.find_unique_graphs is False
+        assert args.save_events is False
+
+    # Test 2: Incorrect file extension for config file
+    with tempfile.NamedTemporaryFile(suffix=".incorrect") as tmp_file:
+        with pytest.raises(ValidationError):
+            args = OtelToPVArgs(config_file=tmp_file.name, command=command)
+
+    # Test 3: Check error handling with save_events=True, command=otel2puml
+    if command == "otel2puml":
+        with tempfile.NamedTemporaryFile(suffix=".yaml") as tmp_file:
+            with pytest.raises(ValidationError):
+                args = OtelToPVArgs(
+                    config_file=tmp_file.name,
+                    command="otel2puml",
+                    save_events=True,
+                )
+
+    # Test 4: Non-existant file
+    with pytest.raises(ValidationError):
+        OtelToPVArgs(config_file="nonexistent.yaml", command="otel2puml")
+
+
+def test_pv_to_puml_args() -> None:
+    """Tests for the pydantic model PvToPumlArgs"""
+
+    # Test 1: Valid filepaths
+    with tempfile.NamedTemporaryFile(
+        suffix=".json"
+    ) as tmp_file1, tempfile.NamedTemporaryFile(suffix=".json") as tmp_file2:
+        file_paths = [tmp_file1.name, tmp_file2.name]
+        args = PvToPumlArgs(file_paths=file_paths)
+        assert args.file_paths
```

Comment:
this isn't a boolean your checking so need to explicitly say what your checking for when you test